pyomo/core/expr/visitor.py

```diff
@@ -1244,9 +1245,13 @@ def visiting_potential_leaf(self, node):
                 # opportunity to map the error to a NonConstant / Fixed
                 # expression error
                 if not node.is_fixed():
-                    raise NonConstantExpressionError()
+                    raise NonConstantExpressionError(
+                        f"{node} ({type(node).__name__}) is not fixed"
+                    )
                 if not node.is_constant():
-                    raise FixedExpressionError()
+                    raise FixedExpressionError(
+                        f"{node} ({type(node).__name__}) is not constant"
+                    )
```

Comment:
I was confused, too: that is why I added the text description.  I thought about changing it, but it has been that way since PR #467, so I was hesitant to change a "public" API (even one that is not used).  I believe that the original meaning was that if you are evaluating an expression that is supposed to be constant and run across a fixed variable, then you get a `FixedExpressionError`.  Otherwise, if you run across a free variable, then you get a generic `NonConstantExpressionError`.

IMHO, the long-term solution is to remove that walker.